homeassistant/components/fritz/common.py

```diff
@@ -42,6 +43,36 @@
 _LOGGER = logging.getLogger(__name__)
 
 
+def _is_tracked(mac: str, current_devices: ValuesView) -> bool:
+    """Check if device is already tracked."""
+    for tracked in current_devices:
+        if mac in tracked:
+            return True
+    return False
+
+
+def device_filter_out_from_trackers(
+    mac: str,
+    device: FritzDevice,
+    pref_disable_new_entities: bool,
+    current_devices: ValuesView,
+) -> bool:
+    """Check if device should be filtered out from trackers."""
+    reason: str | None = None
+    if device.ip_address == "":
+        reason = "Missing IP"
+    elif _is_tracked(mac, current_devices):
+        reason = "Already tracked"
+    elif pref_disable_new_entities:
```

Comment:
The integration shouldn't need to handle the system option. It will be handled by the core in the config entries, entity registry and entity platform helper.

Only not already enabled entities are meant to be filtered out with the system option, not existing enabled entities.

Reply:
Unfortunately, with previous code at each restart all device_trackers are added back. So basically `pref_disable_new_entities` has no effect at all.

At the same time I agree that we have a issue with this PR (dunno why I didn't spot it before), that I and @AaronDavidSchneider discussed this morning.
He is going to provide a quick fix for that.

Simone

Reply:
After discussing with @MartinHjelmare on discord, I got the point that current system is not built to avoid adding all entities to the registry; the system option is meant to avoid adding the entities to the state machine (this means that entities are still created but disabled).

We need to achive a decision in the architecture discussion in order to handle this task properly.

I also suggest to clarify the message for users:

![image](https://user-images.githubusercontent.com/57354320/135353932-6bc79f56-8ec9-40e1-9b35-e72881b1651e.png)